components/lib/state/src/test/java/mozilla/components/lib/state/ext/FragmentKtTest.kt

```diff
@@ -261,4 +268,32 @@ class FragmentKtTest {
         assertEquals(25, receivedValue)
         latch = CountDownLatch(1)
     }
-}
+
+    @Test
+    fun `consumeFlow - creates flow synchronously`() {
```

Comment:
I tried for a long time to write a test that reliably replays the real-world problem, but found that's only possible with a bigger refactoring. We'd need to change the lifecycle of the fragment in between calling `store.flow` and our `viewScope.launch` which I can only do if I use different dispatchers/scopes for the test and then technically it's different than in the app again so I decided for this simpler test that just makes sure we create the flow outside the coroutine.